# HG changeset patch
# User Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date 1655702496 0
# Node ID 82da2b52e0b563ec60d00f71fd97114446392ecc
# Parent  abc325a81822df4b784e9536c3d7250824e1909d
Bug 1773879 - [devtools] Fix Copy all Message context menu action in Browser Console. r=jdescottes.

We don't have a toolbox in the Browser Console, so we don't
need to have a Provider for its store.
A simple test is added, just checking that the action was
triggered.

Differential Revision: https://phabricator.services.mozilla.com/D149593

diff --git a/devtools/client/webconsole/utils/context-menu.js b/devtools/client/webconsole/utils/context-menu.js
--- a/devtools/client/webconsole/utils/context-menu.js
+++ b/devtools/client/webconsole/utils/context-menu.js
@@ -333,17 +333,23 @@ async function getUnvirtualizedConsoleOu
     });
 
     ReactDOM.render(
       createElement(
         Provider,
         {
           store: webConsoleWrapper.getStore(),
         },
-        createElement(ToolboxProvider, { store: toolbox.store }, consoleOutput)
+        toolbox
+          ? createElement(
+              ToolboxProvider,
+              { store: toolbox.store },
+              consoleOutput
+            )
+          : consoleOutput
       ),
       singleUseElement,
       () => {
         resolve(getElementText(singleUseElement));
         singleUseElement.remove();
         ReactDOM.unmountComponentAtNode(singleUseElement);
       }
     );

